infra/general/kString.class.php

```diff
@@ -440,4 +440,9 @@ public static function stripInvalidUrlChars($url)
 		return str_replace(array('?', '|', '*', '\\', '/' , '>' , '<', '&', '[', ']',' '), '_', $url);
 	}
 
+	public static function keepOnlyValidUrlChars($url)
+	{
+	    return preg_replace('/[^A-Za-z0-9\-._~!$&()*+,;=:@]/', '_', $url);
```

Comment:
@MosheMaorKaltura I tested the change of using keepOnlyValidUrlChars on my own environment and verified it replaced bad characters with underscore and that the download was successful.
About tantan's note, I removed the unnecessary slashes and tested that preg_replace still replacing characters not in that list.